### PR TITLE
Draw sample in background - frontend

### PR DESCRIPTION
**Pull request by jonahkagan**
There are two places that we create a new round - the audit setup review screen (for round 1) and the status box (for subsequent rounds). In order to support both of these places creating rounds and waiting for the sample drawing to complete in the background, we check if the sample is being drawn before rendering any of the the audit view, and show a spinner if the background task is still in progress.

Technical changes:
- Consolidated round creation in the useRoundsAuditAdmin hook (startNewRounds function), since we need to reload the rounds once the new round is created
- Since startNewRounds requires sample sizes as numbers, simplified the Review screen to use a numeric input

### CODE REVIEW

File: client/src/components/MultiJurisdictionAudit/useSetupMenuItems/_mocks/index.ts [hunk 1]
```diff
@@ -123,12 +123,7 @@ export const auditSettings: {
 }
 
 export const roundMocks: {
-  [key in
-    | 'empty'
-    | 'singleIncomplete'
-    | 'twoIncomplete'
-    | 'singleComplete'
-    | 'needAnother']: IRound[]
+  [key: string]: IRound[]
```
Comment (MorganLove) [inline client/src/components/MultiJurisdictionAudit/useSetupMenuItems/_mocks/index.ts:126]: I find the explicit defining of each state to be useful in IDE autocompletion, personally.

File: client/src/components/testUtilities.tsx [hunk 2]
```diff
@@ -128,9 +128,10 @@ export const withMockFetch = async (
     }
 
     if (expectedRequest) {
+      const requestIndex = requests.length - requestsLeft.length - 1
       // eslint-disable-next-line no-console
       console.error(
-        'Expected fetch request:\n',
+        `Expected fetch request (${requestIndex}):\n`,
```
Comment (MorganLove) [inline client/src/components/testUtilities.tsx:134]: Nice!

**Review by MorganLove — APPROVED** [review]
Looks good! One suggestion that's not blocking, that's all.

File: client/src/components/MultiJurisdictionAudit/useSetupMenuItems/_mocks/index.ts (hunk 1 above)
Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/useSetupMenuItems/_mocks/index.ts:126]: Ohhh cool I was wondering why they were there. Thanks for the tip. I'll put them back. I wish the IDEs were smart enough to figure it out without the type annotation...

**Pull request merged**